rfcs/text/0000-roadmaps+planning.md

```diff
@@ -0,0 +1,431 @@
+
+***Next Release Milestone 1 Week 3***. Further refinements for Milestone 2 are completely modeled. All Milestone 1 issues that will slip to Milestone 2 MUST be identified.
+
+***Next Release Milestone 2 Week 3***. Further refinements for Milestone 3 are completely modeled. All Milestone 2 issues that will slip to Milestone 3 MUST be identified.
+
+***Next Release Milestone 3 Week 3***.
+All Milestone 3 issues that will slip to the next Release MUST be identified.
+
+#### Tracking Release Progress
+Product Owners for each Objective MUST ensure that their Epics and child issues are kept up-to-date with current status.
+
+Technical Leads MUST drive implementations of prioritized Epics forward and communicate blockers or changes in schedule if they arise.
+
+Over-communication is strongly RECOMMENDED.
+
+***Every two weeks***. The Release Shepherd facilitates a one hour meeting for Product Owners, User Experience, and Technical Leads to review the progress of the Release in ZenHub and address emerging issues.
+
+The Release Shepherd MUST send a status update to the Project Leads on the DCP PM mailing list that captures action items for participants, documents slips in schedule, and/or requests clarification as needed.
+
+As required, the Release Shepherd SHOULD schedule time on the next DCP PM call to address issues pending resolution. The Project Leads MUST never be surprised by the *state of play*.
+
+***At Milestone Completion***. The Project Leads MUST update Oversight on the progress of the Release to ensure appropriate accountability.
+
+#### Modeling progress with ZenHub Pipelines [*dcp* repo]
+
+New items in the *dcp* repo appear in the ZenHub *New* pipeline and MUST be triaged into the *Product Backlog* pipeline, the *Icebox* pipeline, or the *Closed* pipeline by the Release Shepherd or Product Owners.
+
+The ZenHub *Icebox* pipeline SHOULD be used with great restraint in the *dcp* repo. Its explicit purpose is to queue up potential futures from the DCP roadmap such as pending DevSecOps issues. Individual component repo(s)  MAY use the *Icebox* pipeline in a manner of their own choosing.
+
+No issues in the *dcp* repo MUST be assigned to the *Epic* or *Sprint Backlog* pipelines. Individual component repo(s) MAY use these pipelines in a manner of their own choosing.
+
+At the level of the *dcp* repo, a subset of ZenHub pipelines are used to track progress. All Epics or issues in a Release are initially assigned to the *Product Backlog* pipeline. Once the Epic (or any of its children) is started, then its *dcp* issue is moved to the *In Progress* pipeline. When all the children of the Epic are in the *Closed* pipeline, then the *dcp* issue is moved to the *Closed* pipeline.
+
+#### Modeling progress with ZenHub Pipelines [component repo]
+
+When an issue is started, it is moved to the *In Progress* pipeline.
+
+When a pull request is filed and in review for an *In Progress* item, the issue is moved to the *Review* pipeline.
```

Comment:
As @hannes-ucsc has said elsewhere, please do not remove the other existing pipelines. These are being used by our teams in ways that meet local requirements.

Reply:
There's no intention to remove or rename pipelines without community consensus. The intent is to define a minimal and common pipeline workflow so the Product Owners and Technical Leads can understand the progress of an epic when reviewing the top-level ZenHub board:

<img width="715" alt="Screen Shot 2019-08-08 at 10 18 46 AM" src="https://user-images.githubusercontent.com/3644743/62723647-0597ac00-b9c6-11e9-90dd-a11c37591d17.png">

At the very least, we need to have a common understanding on which pipelines signify _In Progress_ and when the child issues are _Done Done_ so the top-level epic can be closed appropriately. 

There's also a new ZenHub feature that allows repo(s) to belong to multiple workspaces with [customized workflows](https://www.zenhub.com/blog/announcing-zenhub-workspaces-personalized-workflows-for-every-team/).

Reply:
I'd add an explicit statement to this RFC about which columns would be removed.

Reply:
Based on the discussion on the August 22 PM call - the proposed pipelines for the DCP workspace are:
* _New_
* _Product Backlog_
* _Icebox_
* _In Progress_
* _Closed_ (meaning done-done in production).

Again, no changes to ZenHub pipelines will be made without consensus and preparation.